### ISSUE TITLE
Bug when not full size.

### ISSUE THREAD

**Issue by mikes222** [umano/AndroidSlidingUpPanel#809]
When using slidinguppanel in a part of the screen there is an issue. The following structure: 

    <?xml version="1.0" encoding="utf-8"?>
    <LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
        xmlns:sothree="http://schemas.android.com/apk/res-auto"
        android:layout_width="match_parent"
        android:layout_height="match_parent"
        android:orientation="horizontal">
    
        <LinearLayout
            android:layout_width="0dp"
            android:layout_height="match_parent"
            android:layout_weight="1">
    
            <com.mschwartz.bsnstemplate.ui.slidinguppanel.SlidingUpPanelLayout
                android:id="@+id/slidinguppanel"
                android:layout_width="match_parent"
                android:layout_height="match_parent"
                android:gravity="bottom"
                android:orientation="vertical"
                sothree:umanoDragView="@+id/draghere"
                sothree:umanoOverlay="true"
                sothree:umanoPanelHeight="80dp"
                sothree:umanoShadowHeight="0dp">
    ...
            </com.mschwartz.bsnstemplate.ui.slidinguppanel.SlidingUpPanelLayout>
    
        </LinearLayout>
    
        <LinearLayout
            android:layout_width="0dp"
            android:layout_height="match_parent"
            android:layout_weight="1">
    
            <TextView
                android:layout_width="match_parent"
                android:layout_height="match_parent"
                android:gravity="center"
                android:text="right side of the screen" />
        </LinearLayout>
    </LinearLayout>

will produce an exception:

> java.lang.IllegalStateException: Width must have an exact value or MATCH_PARENT
>                                                             at com.sothree.slidinguppanel.SlidingUpPanelLayout.onMeasure(SlidingUpPanelLayout.java:733)
> 

If I remove the 4 lines from code

        if (widthMode != MeasureSpec.EXACTLY && widthMode != MeasureSpec.AT_MOST) {
            throw new IllegalStateException("Width must have an exact value or MATCH_PARENT");
        } else if (heightMode != MeasureSpec.EXACTLY && heightMode != MeasureSpec.AT_MOST) {
            throw new IllegalStateException("Height must have an exact value or MATCH_PARENT");
        }

everything runs like a charm. 

Are there any side effects or can we simple remove the 4 lines?

Regards

**Comment by kunalsharma** [comment]
I am facing the same issue. Are you able to find solution of this issue

**Comment by mikes222** [comment]
@kunalsharma , of course. I just removed the 4 lines and it works since then. See also https://github.com/mikes222/BsnsTemplate/tree/master/app/src/main/java/com/mschwartz/bsnstemplate/ui/slidinguppanel
